src/asset-management/AaveManager.sol

```diff
@@ -53,18 +49,17 @@ contract AaveManager is IAssetManager, Owned(msg.sender), ReentrancyGuard {
     //////////////////////////////////////////////////////////////////////////*/
 
     /// @dev returns the balance of the token managed by various markets in the native precision
-    function getBalance(ReservoirPair aOwner, ERC20 aToken) external view returns (uint104 rTokenBalance) {
+    function getBalance(ReservoirPair aOwner, ERC20 aToken) external view returns (uint256 rTokenBalance) {
         return _getBalance(aOwner, aToken);
     }
 
-    function _getBalance(ReservoirPair aOwner, ERC20 aToken) private view returns (uint104 rTokenBalance) {
+    function _getBalance(ReservoirPair aOwner, ERC20 aToken) private view returns (uint256 rTokenBalance) {
```

Comment:
I have reviewed AaveManager as a whole and concluded that changing the types to `uint256` will not cause any detrimental effects within the manager itself. 

The manager is self-contained, we just have to be careful when the pair queries balances from the manager

Reply:
Okay but how do users get their money back if `_getBalance` returns `type(uint256).max`, is there a way to lower the balances?

Reply:
realistically to hit `type(uint256).max`, the entire supply of the ERC20 token would have to be here, managed by our asset manager. So that wouldn't happen right, you agree? 

Putting this problem in the more likely scenario of the balance exceeding `uint104` (the entire reserves being managed by AM), the AM admin can call `adjustManagement` to get it under, cuz it would make `token0/1Managed` overflow `uint104` as well. But, that still doesn't solve the problem cuz after redeeming, it would make the reserves exceed `uint104` as well. I'll write a test case to demonstrate this problem first. 

It seems like we need a solution similar to `skim` (for `token0/1Managed` instead of reserves), to act as a release valve when the amount gets crazy.

Reply:
Yeah, right now this is a major/critical vuln

Reply:
wrote a test case to demonstrate this scenario happening. 

Still contemplating what's the best solution for it

Reply:
will address this with a solution in the subsequent PR

Reply:
Can resolve if this is captured in a separate issue or the PR is started

Reply:
will fix in #156